test: TestHeadCompactionWhileScraping is flaky

## What happened?

`TestHeadCompactionWhileScraping` is flaky and can fail in CI with "Condition never satisfied".

Example failure: https://github.com/prometheus/prometheus/actions/runs/21449194089/job/61773140350?pr=17740

```
--- FAIL: TestHeadCompactionWhileScraping (0.00s)
    --- FAIL: TestHeadCompactionWhileScraping/0 (15.12s)
        Error Trace:    /__w/prometheus/prometheus/cmd/prometheus/main_test.go:848
        Error:          Condition never satisfied
```

## Root cause

The test waits for at least 3 compactions to occur within the `require.Eventually` timeout:

```go
// Wait for some compactions to run
compactions, err := getMetricValue(t, bytes.NewReader(metrics), model.MetricTypeCounter, "prometheus_tsdb_compactions_total")
if err != nil {
    return false
}
if compactions < 3 {
    return false
}
```

This is timing-dependent and can fail in slow CI environments where compactions may not complete quickly enough.

## Possible fixes

1. Increase the timeout for `require.Eventually`
2. Reduce the number of required compactions
3. Use shorter `min-block-duration` to trigger compactions faster
4. Mark the test with appropriate skip conditions for slow environments

--- a/cmd/prometheus/main_test.go
+++ b/cmd/prometheus/main_test.go
@@ -953,6 +953,9 @@ runtime:
 // concurrently with a scrape does not trigger the data race described in
 // https://github.com/prometheus/prometheus/issues/16490.
 func TestHeadCompactionWhileScraping(t *testing.T) {
+	if runtime.GOOS == "windows" { // No race detector on Windows.
+		t.SkipNow()
+	}
 	t.Parallel()
 
 	// To increase the chance of reproducing the data race
@@ -983,6 +986,7 @@ scrape_configs:
 				port,
 				fmt.Sprintf("--storage.tsdb.path=%s", tmpDir),
 				"--storage.tsdb.min-block-duration=100ms",
+				"--scrape.discovery-reload-interval=100ms", // Reduce initial wait time from default of 5 seconds.
 			)
 			require.NoError(t, prom.Start())
 
@@ -1020,6 +1024,7 @@ scrape_configs:
 				require.Zero(t, failures)
 				return true
 			}, 15*time.Second, 500*time.Millisecond)
+			prom.Process.Signal(os.Interrupt)
 		})
 	}
 }

